### PR TITLE
Update authors in CITATION.cff and maintainer list

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Description

Address #3412

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Should we update `technical_steering_committee.md` too?

File: docs/source/contribution/technical_steering_committee.md [hunk 1]
```diff
@@ -56,28 +56,23 @@ We look for commitment markers who can do the following:
 | Name                                                     | Organisation                                                                            |
```
Comment (astrojuanlu) [inline docs/source/contribution/technical_steering_committee.md:56]: Do we need TSC votes to _remove_ members from the TSC?

cc @yetudada

**Review by astrojuanlu — APPROVED** [review]
I'm +1 but please just in case we need TSC consensus for this, add your approving reviews here and let's not rush merging

File: docs/source/contribution/technical_steering_committee.md (hunk 1 above)
Comment (stichbury) [reply docs/source/contribution/technical_steering_committee.md:56]: We haven't in the past. The charter is a PDF in the root of the repo but I scanned it and it wasn't clear on leaving (only that 2/3 should approve changes to the charter). For QB members, I would say that if someone leaves the team they are voluntarily leaving as a TSC member, unless they explicitly commit to continue contributing and do so by regularly through meeting attendance and/or code contributions/PR reviews. 

It's worth asking each person if they intend to do that.

**Review by stichbury — APPROVED** [review]
Approving, thanks @AhdraMeraliQB

**Review by DimedS — APPROVED** [review]

**Review by deepyaman — APPROVED** [review]
LGTM, but agree with @stichbury that QB members generally join with the expectation that they're on TSC until they leave the team, and we don't practically need a full vote unless they say otherwise. Think this works until somebody complains. :)

**Review by merelcht — CHANGES_REQUESTED** [review]
I think this needs to be discussed in more detail. The members list that would remain after this PR is inconsistent between `CITATION.cff` and the `technical_steering_committee.md`. I'm questioning also why @datajoely  is suddenly removed, since nothing changed in his position? I'd also be less quick to remove @tynandebold since he did say to me he's planning on continuing to work on Kedro.

**Review by tynandebold — CHANGES_REQUESTED** [review]
As Merel said, I do plan on continuing to contribute, and have still been reviewing smaller PRs and opening others for the Kedro website.

**Comment by AhdraMeraliQB** [comment]
Closing to discuss in the new year

**Comment by datajoely** [comment]
Yeah I'd like to stick around! 


I do think, however, we need to revisit the idea of a more consolidated TSC. We should start prioritise LFAI graduation in the 2024 and I believe that means revisiting the QB dev == automatic TSC membership status quo.

**Pull request closed without merging**